src/unix/bsd/apple/mod.rs

```diff
@@ -640,6 +634,133 @@ cfg_if! {
 
 cfg_if! {
     if #[cfg(feature = "extra_traits")] {
+        impl PartialEq for kevent {
+            fn eq(&self, other: &kevent) -> bool {
+                self.ident == other.ident
+                    && self.filter == other.filter
+                    && self.flags == other.flags
+                    && self.fflags == other.fflags
+                    && self.data == other.data
+                    && self.udata == other.udata
+            }
+        }
+        impl Eq for kevent {}
+        impl ::fmt::Debug for kevent {
+            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
+                f.debug_struct("kevent")
+                    .field("ident", unsafe { &self.ident })
```

Comment:
So instead of 

```rust
. field("ident", unsafe { &self.ident })
```

I think the simplest solution would be to write:

```rust
// unnecessary copy but ¯\_(ツ)_/¯ 
let ident = self.ident;  
// ... 
. field("ident", &ident)
```

I'm not sure if all field types are copy (we do have large arrays), but I hope so.

Reply:
> unnecessary copy but ¯\_(ツ)_/¯

There's nothing unnecessary about it. `.field` will perform an aligned load from this address, we need to put stuff into an aligned address.